explorer/e2e/testFunctions.ts

```diff
@@ -207,4 +215,208 @@ export async function testPreSelectedColumns(
   }
 }
 
+export const filterRegex = (filter: string): RegExp =>
+  new RegExp(filter + "\\s+\\([0-9]+\\)\\s*");
+
+export async function testFilterPresence(
+  page: Page,
+  tab: TabDescription,
+  filters: string[]
+): Promise<void> {
+  // Goto the selected tab
+  await page.goto(tab.url);
+  await expect(page.getByRole("tab").getByText(tab.tabName)).toBeVisible();
+  for (const filter of filters) {
+    // Check that each filter is visible and clickable
+    await expect(page.getByText(filterRegex(filter))).toBeVisible();
+    await page.getByText(filterRegex(filter)).click();
+    await expect(page.getByRole("checkbox").first()).toBeVisible();
+    await expect(page.getByRole("checkbox").first()).not.toBeChecked();
+    // Check that clicking out of the filter menu causes it to disappear
+    await page.locator("body").click();
+    await expect(page.getByRole("checkbox")).toHaveCount(0);
+  }
+}
+
+export const getNamedFilterButtonLocator = (
+  page: Page,
+  filterName: string
+): Locator => {
+  return page
+    .getByRole("button")
+    .filter({ has: page.getByRole("checkbox"), hasText: filterName });
+};
+export const getFirstFilterButtonLocator = (page: Page): Locator => {
+  return page
+    .getByRole("button")
+    .filter({ has: page.getByRole("checkbox") })
+    .first();
+};
+
+export async function testFilterPersistence(
+  page: Page,
+  testFilter: string,
+  tabOrder: TabDescription[]
+): Promise<void> {
+  // Start on the first tab in the test order (should be files)
+  await page.goto(tabOrder[0].url);
+  // Select the first checkbox on the test filter
+  await page.getByText(filterRegex(testFilter)).click();
+  const filterToSelectLocator = await getFirstFilterButtonLocator(page);
+  await expect(filterToSelectLocator.getByRole("checkbox")).not.toBeChecked();
+  await filterToSelectLocator.getByRole("checkbox").click();
+  const filterNameMatch = (await filterToSelectLocator.innerText()).match(
+    /\.\S*/
+  );
+  if (filterNameMatch == null) {
+    test.fail();
```

Comment:
Would it be possible to return a boolean here and call `test.fail` in the calling code? This matches the pattern in [testFilterCount](https://github.com/DataBiosphere/data-browser/pull/4071/files#diff-faefc5cfc7d7c7283d514ad866678143558aefea3eccb636b8bb7a1ae4abbae7R307).